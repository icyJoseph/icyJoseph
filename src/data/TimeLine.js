export default [
  {
    title: "April, 2018",
    meta: "CodeJam",
    fill: "#273B7A",
    backgroundFill: "tomato",
    description: `The description goes here. The quick brown fox, jumps over the lazy dog
      The description goes here. The quick brown fox, jumps over the lazy dog`
  },
  {
    title: "March, 2018",
    meta: "Hash Code",
    fill: "#273B7A",
    backgroundFill: "tomato",
    description: `The description goes here. The quick brown fox, jumps over the lazy dog`
  },
  {
    title: "January, 2018",
    meta: "React Certificate",
    fill: "#273B7A",
    backgroundFill: "tomato",
    description: `The description goes here. The quick brown fox, jumps over the lazy dog`
  },
  {
    title: "January, 2018",
    meta: "Volvo Group",
    fill: "#C92F00",
    backgroundFill: "tomato",
    description: `The description goes here. The quick brown fox, jumps over the lazy dog`
  },
  {
    title: "November, 2017",
    meta: "Open Hack Winner",
    fill: "#273B7A",
    backgroundFill: "tomato",
    description: `The description goes here. The quick brown fox, jumps over the lazy dog`
  },
  {
    title: "October, 2016",
    meta: "Consultant",
    fill: "#292929",
    backgroundFill: "#5159BF",
    description: `The description goes here. The quick brown fox, jumps over the lazy dog`
  },
  {
    title: "June, 2016",
    meta: "M.Sc. in Science with Major in Entepreneurship and Business Design",
    fill: "#292929",
    backgroundFill: "tomato",
    description: `The description goes here. The quick brown fox, jumps over the lazy dog`
  },
  {
    title: "January, 2016",
    meta: "CTO, Co-founder",
    fill: "#F0DB4F",
    backgroundFill: "tomato",
    description: `The description goes here. The quick brown fox, jumps over the lazy dog`
  },
  {
    title: "March, 2013",
    meta: "Project Engineer",
    fill: "#273B7A",
    backgroundFill: "tomato",
    description: `The description goes here. The quick brown fox, jumps over the lazy dog`
  },
  {
    title: "December, 2012",
    meta: "B.Sc. in Science with Major in Electronics Engineering",
    fill: "#273B7A",
    backgroundFill: "tomato",
    description: `The description goes here. The quick brown fox, jumps over the lazy dog`
  }
];
