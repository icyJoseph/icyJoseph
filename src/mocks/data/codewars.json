{
  "username": "icyJoseph",
  "name": "Joseph",
  "honor": 663,
  "clan": "Evolve Technology Sweden",
  "leaderboardPosition": 38381,
  "skills": null,
  "ranks": {
    "overall": { "rank": -4, "name": "4 kyu", "color": "blue", "score": 1158 },
    "languages": {
      "rust": { "rank": -4, "name": "4 kyu", "color": "blue", "score": 1158 }
    }
  },
  "codeChallenges": { "totalAuthored": 0, "totalCompleted": 28 }
}
