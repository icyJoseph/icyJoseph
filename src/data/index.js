import React from "react";
import Loadable from "react-loadable";
import Spinner from "../components/Loading/Spinner";
import Loading from "../logos/Loading";
import TimeLine from "./TimeLine";

export default [
  {
    id: 0,
    title: "About Me",
    meta: "Joseph",
    Component: Loadable({
      loader: () => import("../logos/Me"),
      loading() {
        return <Loading />;
      }
    }),
    fill: "#273B7A",
    background: "#193549",
    backgroundFill: "tomato",
    Content: Loadable({
      loader: () => import("../components/About"),
      loading() {
        return <Spinner />;
      }
    })
  },
  {
    id: 1,
    title: "ReactJS",
    meta: "Certified",
    Component: Loadable({
      loader: () => import("../logos/ReactLogo"),
      loading() {
        return <Loading />;
      }
    }),
    fill: "#292929",
    background: "#193549",
    backgroundFill: "tomato",
    Content: Loadable({
      loader: () => import("../components/Portfolio"),
      render(loaded, props) {
        let Component = loaded.default;
        return <Component {...props} meta="react" type="repos" />;
      },
      loading() {
        return <Spinner />;
      }
    })
  },
  {
    id: 2,
    title: "JavaScript",
    meta: "Favorite Language",
    background: "#193549",
    Component: Loadable({
      loader: () => import("../logos/JavaScriptLogo"),
      loading() {
        return <Loading />;
      }
    }),
    fill: "#F0DB4F",
    backgroundFill: "tomato",
    Content: Loadable({
      loader: () => import("../components/Recast"),
      loading() {
        return <Spinner />;
      }
    })
  },
  {
    id: 3,
    title: "Redux",
    meta: "Framework",
    Component: Loadable({
      loader: () => import("../logos/Redux"),
      loading() {
        return <Loading />;
      }
    }),
    fill: "#292929",
    background: "#193549",
    backgroundFill: "#5159BF",
    Content: Loadable({
      loader: () => import("../components/Portfolio"),
      render(loaded, props) {
        let Component = loaded.default;
        return <Component {...props} meta="redux" type="repos" />;
      },
      loading() {
        return <Spinner />;
      }
    })
  },
  {
    id: 4,
    title: "algorithms",
    meta: "Problem Solving",
    Component: Loadable({
      loader: () => import("../logos/Problem"),
      loading() {
        return <Loading />;
      }
    }),
    fill: "#273B7A",
    background: "#193549",
    backgroundFill: "tomato",
    Content: Loadable({
      loader: () => import("../components/Portfolio"),
      render(loaded, props) {
        let Component = loaded.default;
        return <Component {...props} meta="algorithms" type="solutions" />;
      },
      loading() {
        return <Spinner />;
      }
    })
  },
  {
    id: 5,
    title: "experience",
    meta: "My career",
    Component: Loadable({
      loader: () => import("../logos/Experience"),
      loading() {
        return <Loading />;
      }
    }),
    fill: "#273B7A",
    background: "#193549",
    backgroundFill: "tomato",
    Content: Loadable({
      loader: () => import("../components/TimeLine"),
      render(loaded, props) {
        let Component = loaded.default;
        return <Component {...props} experience data={TimeLine} />;
      },
      loading() {
        return <Spinner />;
      }
    })
  },
  {
    id: 6,
    title: "python",
    meta: "Code as a hobby",
    Component: Loadable({
      loader: () => import("../logos/PythonLogo"),
      loading() {
        return <Loading />;
      }
    }),
    fill: "#273B7A",
    background: "#193549",
    backgroundFill: "tomato",
    Content: Loadable({
      loader: () => import("../components/Portfolio"),
      render(loaded, props) {
        let Component = loaded.default;
        return <Component {...props} meta="python" />;
      },
      loading() {
        return <Spinner />;
      }
    })
  },
  {
    id: 7,
    title: "hackathon",
    meta: "Value Creation",
    Component: Loadable({
      loader: () => import("../logos/Rocket"),
      loading() {
        return <Loading />;
      }
    }),
    fill: "#273B7A",
    background: "#193549",
    backgroundFill: "tomato",
    Content: Loadable({
      loader: () => import("../components/Portfolio"),
      render(loaded, props) {
        let Component = loaded.default;
        return <Component {...props} meta="hackathon" type="hacks" />;
      },
      loading() {
        return <Spinner />;
      }
    })
  },
  {
    id: 8,
    title: "awards",
    meta: "Acknowledgments",
    Component: Loadable({
      loader: () => import("../logos/Award"),
      loading() {
        return <Loading />;
      }
    }),
    fill: "#C92F00",
    backgroundFill: "tomato",
    Content: Loadable({
      loader: () => import("../components/Common"),
      render(loaded, props) {
        let Component = loaded.default;
        return <Component {...props} title={"awards"} />;
      },
      loading() {
        return <Spinner />;
      }
    })
  },
  {
    id: 9,
    title: "currently",
    meta: "Life as a React Developer",
    Component: Loadable({
      loader: () => import("../logos/Report"),
      loading() {
        return <Loading />;
      }
    }),
    fill: "#273B7A",
    backgroundFill: "tomato",
    Content: Loadable({
      loader: () => import("../components/Common"),
      render(loaded, props) {
        let Component = loaded.default;
        return <Component {...props} title={"currently"} />;
      },
      loading() {
        return <Spinner />;
      }
    })
  }
];
