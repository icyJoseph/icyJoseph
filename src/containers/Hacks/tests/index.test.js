import React from "react";
import { shallow } from "enzyme";
import { Hacks } from "../";
import data from "../../../data";

describe("HomeDesktopGrid", () => {
  const closeDrawer = jest.fn();
  const openDrawer = jest.fn();
  const props = {
    visibility: false,
    contentId: 1,
    closeDrawer,
    openDrawer,
    data
  };
  const wrapper = shallow(<Hacks {...props} />);
  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
