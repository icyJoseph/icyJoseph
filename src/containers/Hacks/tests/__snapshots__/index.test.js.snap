// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeDesktopGrid renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Hacks
    closeDrawer={[MockFunction]}
    contentId={1}
    data={
      Array [
        Object {
          "Component": [Function],
          "Content": [Function],
          "background": "#193549",
          "backgroundFill": "tomato",
          "fill": "#273B7A",
          "id": 0,
          "meta": "Joseph",
          "title": "About Me",
        },
        Object {
          "Component": [Function],
          "Content": [Function],
          "background": "#193549",
          "backgroundFill": "tomato",
          "fill": "#292929",
          "id": 1,
          "meta": "Certified",
          "title": "ReactJS",
        },
        Object {
          "Component": [Function],
          "Content": [Function],
          "background": "#193549",
          "backgroundFill": "tomato",
          "fill": "#F0DB4F",
          "id": 2,
          "meta": "Favorite Language",
          "title": "JavaScript",
        },
        Object {
          "Component": [Function],
          "Content": [Function],
          "background": "#193549",
          "backgroundFill": "#5159BF",
          "fill": "#292929",
          "id": 3,
          "meta": "Framework",
          "title": "Redux",
        },
        Object {
          "Component": [Function],
          "Content": [Function],
          "background": "#193549",
          "backgroundFill": "tomato",
          "fill": "#273B7A",
          "id": 4,
          "meta": "Problem Solving",
          "title": "algorithms",
        },
        Object {
          "Component": [Function],
          "Content": [Function],
          "background": "#193549",
          "backgroundFill": "tomato",
          "fill": "#273B7A",
          "hide": true,
          "id": 5,
          "meta": "My career",
          "title": "experience",
        },
        Object {
          "Component": [Function],
          "Content": [Function],
          "background": "#193549",
          "backgroundFill": "tomato",
          "fill": "#273B7A",
          "id": 6,
          "meta": "Code as a hobby",
          "title": "python",
        },
        Object {
          "Component": [Function],
          "Content": [Function],
          "background": "#193549",
          "backgroundFill": "tomato",
          "fill": "#273B7A",
          "id": 7,
          "meta": "Value Creation",
          "title": "hackathon",
        },
        Object {
          "Component": [Function],
          "Content": [Function],
          "backgroundFill": "tomato",
          "fill": "#C92F00",
          "hide": true,
          "id": 8,
          "meta": "Acknowledgments",
          "title": "awards",
        },
        Object {
          "Component": [Function],
          "Content": [Function],
          "backgroundFill": "tomato",
          "fill": "#273B7A",
          "hide": true,
          "id": 9,
          "meta": "Life as a React Developer",
          "title": "currently",
        },
      ]
    }
    openDrawer={[MockFunction]}
    visibility={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <styled.div
          background="codingBackground.jpg"
        />,
        <styled.div
          tint={0.8}
        />,
        <MainTitle
          center={true}
          subtitle="About me and my code"
          title="Coding"
        />,
        <Dock
          clickHandler={[MockFunction]}
          items={
            Array [
              Object {
                "Component": [Function],
                "Content": [Function],
                "background": "#193549",
                "backgroundFill": "tomato",
                "fill": "#273B7A",
                "id": 0,
                "meta": "Joseph",
                "title": "About Me",
              },
              Object {
                "Component": [Function],
                "Content": [Function],
                "background": "#193549",
                "backgroundFill": "tomato",
                "fill": "#292929",
                "id": 1,
                "meta": "Certified",
                "title": "ReactJS",
              },
              Object {
                "Component": [Function],
                "Content": [Function],
                "background": "#193549",
                "backgroundFill": "tomato",
                "fill": "#F0DB4F",
                "id": 2,
                "meta": "Favorite Language",
                "title": "JavaScript",
              },
              Object {
                "Component": [Function],
                "Content": [Function],
                "background": "#193549",
                "backgroundFill": "#5159BF",
                "fill": "#292929",
                "id": 3,
                "meta": "Framework",
                "title": "Redux",
              },
              Object {
                "Component": [Function],
                "Content": [Function],
                "background": "#193549",
                "backgroundFill": "tomato",
                "fill": "#273B7A",
                "id": 4,
                "meta": "Problem Solving",
                "title": "algorithms",
              },
              Object {
                "Component": [Function],
                "Content": [Function],
                "background": "#193549",
                "backgroundFill": "tomato",
                "fill": "#273B7A",
                "id": 6,
                "meta": "Code as a hobby",
                "title": "python",
              },
              Object {
                "Component": [Function],
                "Content": [Function],
                "background": "#193549",
                "backgroundFill": "tomato",
                "fill": "#273B7A",
                "id": 7,
                "meta": "Value Creation",
                "title": "hackathon",
              },
            ]
          }
        />,
        <Drawer
          Content={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "background": "codingBackground.jpg",
          "desktop": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "tint": 0.8,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "center": true,
          "subtitle": "About me and my code",
          "title": "Coding",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "clickHandler": [MockFunction],
          "items": Array [
            Object {
              "Component": [Function],
              "Content": [Function],
              "background": "#193549",
              "backgroundFill": "tomato",
              "fill": "#273B7A",
              "id": 0,
              "meta": "Joseph",
              "title": "About Me",
            },
            Object {
              "Component": [Function],
              "Content": [Function],
              "background": "#193549",
              "backgroundFill": "tomato",
              "fill": "#292929",
              "id": 1,
              "meta": "Certified",
              "title": "ReactJS",
            },
            Object {
              "Component": [Function],
              "Content": [Function],
              "background": "#193549",
              "backgroundFill": "tomato",
              "fill": "#F0DB4F",
              "id": 2,
              "meta": "Favorite Language",
              "title": "JavaScript",
            },
            Object {
              "Component": [Function],
              "Content": [Function],
              "background": "#193549",
              "backgroundFill": "#5159BF",
              "fill": "#292929",
              "id": 3,
              "meta": "Framework",
              "title": "Redux",
            },
            Object {
              "Component": [Function],
              "Content": [Function],
              "background": "#193549",
              "backgroundFill": "tomato",
              "fill": "#273B7A",
              "id": 4,
              "meta": "Problem Solving",
              "title": "algorithms",
            },
            Object {
              "Component": [Function],
              "Content": [Function],
              "background": "#193549",
              "backgroundFill": "tomato",
              "fill": "#273B7A",
              "id": 6,
              "meta": "Code as a hobby",
              "title": "python",
            },
            Object {
              "Component": [Function],
              "Content": [Function],
              "background": "#193549",
              "backgroundFill": "tomato",
              "fill": "#273B7A",
              "id": 7,
              "meta": "Value Creation",
              "title": "hackathon",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "Content": [Function],
          "github": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <styled.div
            background="codingBackground.jpg"
          />,
          <styled.div
            tint={0.8}
          />,
          <MainTitle
            center={true}
            subtitle="About me and my code"
            title="Coding"
          />,
          <Dock
            clickHandler={[MockFunction]}
            items={
              Array [
                Object {
                  "Component": [Function],
                  "Content": [Function],
                  "background": "#193549",
                  "backgroundFill": "tomato",
                  "fill": "#273B7A",
                  "id": 0,
                  "meta": "Joseph",
                  "title": "About Me",
                },
                Object {
                  "Component": [Function],
                  "Content": [Function],
                  "background": "#193549",
                  "backgroundFill": "tomato",
                  "fill": "#292929",
                  "id": 1,
                  "meta": "Certified",
                  "title": "ReactJS",
                },
                Object {
                  "Component": [Function],
                  "Content": [Function],
                  "background": "#193549",
                  "backgroundFill": "tomato",
                  "fill": "#F0DB4F",
                  "id": 2,
                  "meta": "Favorite Language",
                  "title": "JavaScript",
                },
                Object {
                  "Component": [Function],
                  "Content": [Function],
                  "background": "#193549",
                  "backgroundFill": "#5159BF",
                  "fill": "#292929",
                  "id": 3,
                  "meta": "Framework",
                  "title": "Redux",
                },
                Object {
                  "Component": [Function],
                  "Content": [Function],
                  "background": "#193549",
                  "backgroundFill": "tomato",
                  "fill": "#273B7A",
                  "id": 4,
                  "meta": "Problem Solving",
                  "title": "algorithms",
                },
                Object {
                  "Component": [Function],
                  "Content": [Function],
                  "background": "#193549",
                  "backgroundFill": "tomato",
                  "fill": "#273B7A",
                  "id": 6,
                  "meta": "Code as a hobby",
                  "title": "python",
                },
                Object {
                  "Component": [Function],
                  "Content": [Function],
                  "background": "#193549",
                  "backgroundFill": "tomato",
                  "fill": "#273B7A",
                  "id": 7,
                  "meta": "Value Creation",
                  "title": "hackathon",
                },
              ]
            }
          />,
          <Drawer
            Content={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "background": "codingBackground.jpg",
            "desktop": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "tint": 0.8,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "center": true,
            "subtitle": "About me and my code",
            "title": "Coding",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "clickHandler": [MockFunction],
            "items": Array [
              Object {
                "Component": [Function],
                "Content": [Function],
                "background": "#193549",
                "backgroundFill": "tomato",
                "fill": "#273B7A",
                "id": 0,
                "meta": "Joseph",
                "title": "About Me",
              },
              Object {
                "Component": [Function],
                "Content": [Function],
                "background": "#193549",
                "backgroundFill": "tomato",
                "fill": "#292929",
                "id": 1,
                "meta": "Certified",
                "title": "ReactJS",
              },
              Object {
                "Component": [Function],
                "Content": [Function],
                "background": "#193549",
                "backgroundFill": "tomato",
                "fill": "#F0DB4F",
                "id": 2,
                "meta": "Favorite Language",
                "title": "JavaScript",
              },
              Object {
                "Component": [Function],
                "Content": [Function],
                "background": "#193549",
                "backgroundFill": "#5159BF",
                "fill": "#292929",
                "id": 3,
                "meta": "Framework",
                "title": "Redux",
              },
              Object {
                "Component": [Function],
                "Content": [Function],
                "background": "#193549",
                "backgroundFill": "tomato",
                "fill": "#273B7A",
                "id": 4,
                "meta": "Problem Solving",
                "title": "algorithms",
              },
              Object {
                "Component": [Function],
                "Content": [Function],
                "background": "#193549",
                "backgroundFill": "tomato",
                "fill": "#273B7A",
                "id": 6,
                "meta": "Code as a hobby",
                "title": "python",
              },
              Object {
                "Component": [Function],
                "Content": [Function],
                "background": "#193549",
                "backgroundFill": "tomato",
                "fill": "#273B7A",
                "id": 7,
                "meta": "Value Creation",
                "title": "hackathon",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "Content": [Function],
            "github": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
